src/main/java/org/simulator/sbml/SBMLinterpreter.java

```diff
@@ -1220,16 +1195,20 @@ public void init(boolean renewTree, double defaultSpeciesValue, double defaultPa
         }
       }
     }
-    int sizeY = model.getCompartmentCount() + model.getSpeciesCount() + model.getParameterCount() + speciesReferencesInRateRules;
-    if (sizeY != Y.length) {
-      Y = new double[sizeY];
-      compartmentIndexes = new int[Y.length];
-      inConcentrationValues = new boolean[Y.length];
-      symbolIdentifiers = new String[Y.length];
-      conversionFactors = new double[Y.length];
-      Arrays.fill(conversionFactors, 1d);
-    }
 
+    int sizeY = model.getCompartmentCount() + model.getSpeciesCount() + model.getParameterCount() + speciesReferencesInRateRules;
+    Y = new double[sizeY];
+    oldY = new double[sizeY];
+    oldY2 = new double[sizeY];
+    changeRate = new double[sizeY];
+    isAmount = new boolean[sizeY];
+    compartmentIndexes = new int[sizeY];
+    conversionFactors = new double[sizeY];
+    inConcentrationValues = new boolean[sizeY];
+    Arrays.fill(conversionFactors, 1d);
```

Comment:
write this directly under the definition of the conversionFactors! don't split initialization from the definition.